### PR TITLE
necessary changes to fix Open Watcom build

### PR DESCRIPTION
**Pull request by jmalak**
Fix the issue with function order for interrupt routines
Little optimization for STI and CLI instruction to use as inline code by #pragma aux

### CODE REVIEW

**Review by ccawley2011 — APPROVED** [review]
Looks good, thanks. I'll wait for a response to https://github.com/libxmp/xmp-cli/pull/49#issuecomment-1206835155 before merging this, though.

**Comment by jmalak** [comment]
> Looks good, thanks. I'll wait for a response to [libxmp#49 (comment)](https://github.com/libxmp/xmp-cli/pull/49#issuecomment-1206835155) before merging this, though.

It is for you to decide what you will use and how. It is for demonstration how it can be done. Anyway it is specific case.

**Pull request closed without merging**
